### PR TITLE
Eliminate table unpack

### PR DESCRIPTION
**Pull request by rosik**
Unpack isn't jit-friendly

### CODE REVIEW

File: membership.lua [hunk 1]
```diff
@@ -28,39 +26,39 @@ local _sock = nil
...
     local parts = uri_tools.parse(uri)
     if not parts then
-        return nil, nil, 'parse error'
+        return nil, 'parse error'
     end
 
-    local hosts = socket.getaddrinfo(parts.host, parts.service, {family='AF_INET', type='SOCK_DGRAM'})
-    if hosts == nil or #hosts == 0 then
-        return nil, nil, 'getaddrinfo failed'
+    local addrinfo = socket.getaddrinfo(
```
Comment (olegrok) [inline membership.lua:45]: You should check the second value of getaddrinfo.

See https://github.com/tarantool/doc/issues/1431 (Return error message as the second return value.)

If it's empty we should return `'getaddrinfo failed'`

File: membership.lua [hunk 2]
```diff
@@ -85,8 +83,8 @@ end
 
 local function send_message(uri, msg_type, msg_data)
     checks('string', 'string', 'table')
-    local host, port = resolve(uri)
-    if not host then
+    local addr = resolve(uri)
```
Comment (olegrok) [inline membership.lua:86]: You skip error message here. Is it OK?

File: membership.lua [hunk 3]
```diff
@@ -182,16 +180,16 @@ local function send_message(uri, msg_type, msg_data)
...
     return ret and ret > 0
 end
 
 local function send_anti_entropy(uri, msg_type, remote_tbl)
     -- send to `uri` all local members that are not in `remote_tbl`
     -- well, not all actualy, but all that fits into UDP packet
     checks('string', 'string', 'table')
-    local host, port = resolve(uri)
-    if not host then
+    local addr = resolve(uri)
```
Comment (olegrok) [inline membership.lua:191]: Error message is skipped

**Comment by olegrok** [comment]
I'm so sorry for off-topic comments. In general look good.

File: membership.lua (hunk 2 above)
Comment (rosik) [reply membership.lua:86]: Well. Actually I could log it on debug level, but not higher. Anyway, the log will be huge and barely useful. We've never encounter such a need, so let's ignore it.

**Review by olegrok — APPROVED** [review]

**Pull request merged**